diff --git a/dist/src/index.js b/dist/src/index.js
index 8eec91714c82eaba243e87e3a7b90863c8b7710d..b97ab1e24e5cf569d7dded56dab0d3463f76f603 100644
--- a/dist/src/index.js
+++ b/dist/src/index.js
@@ -95437,13 +95437,13 @@ class ExportedNames {
             .map(([key, value]) => {
             if (onlyTyped && !value.type)
                 return;
-            return `${dontAddTypeDef && value.doc ? `\n${value.doc}` : ''}${value.identifierText || key}: ${key}`;
+            return `${dontAddTypeDef && value.doc ? `\n${value.doc}` : ''}${JSON.stringify(value.identifierText || key)}: ${key}`;
         })
             .filter(Boolean);
     }
     createReturnElementsType(names, addDoc = true, forceRequired = false) {
         return names.map(([key, value]) => {
-            const identifier = `${value.doc && addDoc ? `\n${value.doc}` : ''}${value.identifierText || key}${value.required || forceRequired ? '' : '?'}`;
+            const identifier = `${value.doc && addDoc ? `\n${value.doc}` : ''}${JSON.stringify(value.identifierText || key)}${value.required || forceRequired ? '' : '?'}`;
             if (!value.type) {
                 return `${identifier}: typeof ${key}`;
             }
