diff --git a/index.js b/index.js
index e057a73b144ffd050a5f6e84faf8c7bb2989f8b7..a594d31970e697fea13bb0341f178a67a2e350fa 100644
--- a/index.js
+++ b/index.js
@@ -7695,13 +7695,13 @@ class ExportedNames {
             if (onlyTyped && !value.type)
                 return;
             // Important to not use shorthand props for rename functionality
-            return `${dontAddTypeDef && value.doc ? `\n${value.doc}` : ''}${value.identifierText || key}: ${key}`;
+            return `${dontAddTypeDef && value.doc ? `\n${value.doc}` : ''}${JSON.stringify(value.identifierText || key)}: ${key}`;
         })
             .filter(Boolean);
     }
     createReturnElementsType(names, addDoc = true, forceRequired = false) {
         return names.map(([key, value]) => {
-            const identifier = `${value.doc && addDoc ? `\n${value.doc}` : ''}${value.identifierText || key}${value.required || forceRequired ? '' : '?'}`;
+            const identifier = `${value.doc && addDoc ? `\n${value.doc}` : ''}${JSON.stringify(value.identifierText || key)}${value.required || forceRequired ? '' : '?'}`;
             if (!value.type) {
                 return `${identifier}: typeof ${key}`;
             }
diff --git a/index.mjs b/index.mjs
index fee2fa56991870807b280d4debb70e8fd5800b58..21b5d1fee2ebbd5ce0b1d0f13e9926f3605eab14 100644
--- a/index.mjs
+++ b/index.mjs
@@ -7675,13 +7675,13 @@ class ExportedNames {
             if (onlyTyped && !value.type)
                 return;
             // Important to not use shorthand props for rename functionality
-            return `${dontAddTypeDef && value.doc ? `\n${value.doc}` : ''}${value.identifierText || key}: ${key}`;
+            return `${dontAddTypeDef && value.doc ? `\n${value.doc}` : ''}${JSON.stringify(value.identifierText || key)}: ${key}`;
         })
             .filter(Boolean);
     }
     createReturnElementsType(names, addDoc = true, forceRequired = false) {
         return names.map(([key, value]) => {
-            const identifier = `${value.doc && addDoc ? `\n${value.doc}` : ''}${value.identifierText || key}${value.required || forceRequired ? '' : '?'}`;
+            const identifier = `${value.doc && addDoc ? `\n${value.doc}` : ''}${JSON.stringify(value.identifierText || key)}${value.required || forceRequired ? '' : '?'}`;
             if (!value.type) {
                 return `${identifier}: typeof ${key}`;
             }
